api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -170,4 +177,53 @@ export class ScenariosController {
       await this.service.getFeatures(id),
     );
   }
+
+  @ApiOkResponse({
+    type: ScenarioSolutionResultDto,
+  })
+  @ApiImplicitQuery({
+    name: 'best',
+    required: false,
+    type: Boolean,
+  })
+  @ApiImplicitQuery({
+    name: 'most-different',
+    required: false,
+    type: Boolean,
+  })
+  @JSONAPIQueryParams()
+  @Get(`:id/marxan/run/:runId/solutions`)
+  async getScenarioRunSolutions(
+    @Param('id', ParseUUIDPipe) id: string,
+    @Param('runId', ParseUUIDPipe) runId: string,
+    @ProcessFetchSpecification() fetchSpecification: FetchSpecification,
+    @Query('best', ParseBoolPipe) selectOnlyBest?: boolean,
+    @Query('most-different', ParseBoolPipe) selectMostDifferent?: boolean,
```

Comment:
they seem to be mutually exclusive
maybe it's not a problem, though
also a dto could be used here

Reply:
yes, they are exclusive (and returning different DTO). Any idea how to show it in code?

Reply:
honestly I'd create another method and route for it, trying to show it in the code with one method is tricky (especially when you'd like to keep it in the openapi spec) and can be not worth the effort

Reply:
@kgajowy should we remove in a new PR the `best` and `most-different` queries and relative code paths in this controller, or did you want to leave these in to let API clients query for best and most different solutions in either way? (I am fine either way, though I'd prefer to have only one way to query for best and most efficient, via their own endpoint).